### PR TITLE
Feature/us9589 check import functionality changes

### PR DESCRIPTION
**Pull request by timmyg**

### CODE REVIEW

File: crds-mp-tools.Tests/Controllers/DonationControllerTest.cs [hunk 1]
```diff
@@ -117,5 +118,38 @@ public static List<MpProgram> GetListOfPrograms()
                 }
             };
         }
+
+        [TestMethod]
+        public async Task ShouldCreateAScholarship()
+        {
+            var viewModel = GetDonationViewModel();
+
```
Comment (timmyg) [inline crds-mp-tools.Tests/Controllers/DonationControllerTest.cs]: @jpc11623 I know you were trying to write controller tests - what should we do with this one?

File: MinistryPlatform/Stewardship/DonationRepository.cs [hunk 2]
```diff
@@ -16,21 +18,22 @@ public class DonationRepository : MinistryPlatformBase, IDonationRepository
...
         {
             return Observable.Create<MpDonation>(observer =>
             {
                 try
                 {
-                    var authToken = _apiUserRepository.GetApiClientToken("CRDS.Tools.TripFinance");
-                    var result = MinistryPlatformRest.UsingAuthenticationToken(authToken).Create(donation);
+                    var result = MinistryPlatformRest.UsingAuthenticationToken(token).Create(donation);
                     _batchRepository.UpdateBatchCount(result.BatchId);
+                    _batchRepository.UpdateBatchAmount(result.BatchId);
```
Comment (timmyg) [inline MinistryPlatform/Stewardship/DonationRepository.cs]: @andycanterbury @silbermm will this mess up things for trips? I can pass in an optional param

Comment (timmyg) [reply MinistryPlatform/Stewardship/DonationRepository.cs]: (andy said it won't effect their stuff as they were doing transfers, thus creating $0 batches)

File: MinistryPlatform/Enum/PaymentType.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        
+        public static PaymentType GetPaymentType(int paymentTypeId)
+        {
+            foreach (var paymentType in paymentTypes)
+            {
+                if (paymentType.id == paymentTypeId)
+                    return paymentType;
+            }
+
+            return CreditCard;
```
Comment (JoeKer1) [inline MinistryPlatform/Enum/PaymentType.cs]: Should this throw an error? I'm assuming that a default of Credit card is ok, just wanted to make sure.

File: MinistryPlatform/Stewardship/BatchRepository.cs [hunk 4]
```diff
@@ -90,6 +114,29 @@ public int CreateNewTransferBatchAndDeposit(DateTime transferDate)
             return batch.BatchId;
         }
 
+        public int CreateNewNonAchBatch(string batchName, DateTime processDate, int userId)
```
Comment (JoeKer1) [inline MinistryPlatform/Stewardship/BatchRepository.cs]: Should these be IObservable<int>'s like the old methods in here?

File: crds-mp-tools.Tests/Controllers/DonationControllerTest.cs [hunk 5]
```diff
@@ -117,5 +118,6 @@ public static List<MpProgram> GetListOfPrograms()
                 }
             };
         }
+
```
Comment (JoeKer1) [inline crds-mp-tools.Tests/Controllers/DonationControllerTest.cs]: Did these two changes mean to get in here?

File: crds-mp-tools/Services/Donations/DonationService.cs [hunk 6]
```diff
@@ -41,5 +56,65 @@ public DonationViewModel BuildViewModel()
...
+                try
+                {
+                    // Mark the check as exported now, so we don't double-charge a community member.
+                    // If the CreateDonationAndDistributionRecord fails, we'll still consider it exported, but
+                    // it will be in error, and will have to be manually resolved.
+                    check.Exported = true;
+
+                    var donationAndDistribution = new MpDonationAndDistributionRecord
+                    {
+						// TODO this should try to look up donor first
```
Comment (JoeKer1) [inline crds-mp-tools/Services/Donations/DonationService.cs]: spaaaaace

**Pull request merged**